src/allmydata/storage_client_http.py

```diff
@@ -0,0 +1,204 @@
+"""
+HTTP-based storage client.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from future.utils import PY2
+if PY2:
+    from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, list, object, range, str, max, min  # noqa: F401
+
+import attr
+from zope.interface import implementer
+from twisted.internet.defer import inlineCallbacks, returnValue, Deferred, succeed
+
+from .interfaces import (
+    IStorageClientV2, IStorageServer,
+    TestVector, ReadVector, TestWriteVectors,
+    TestVectorOperator, WriteVector,
+)
+
+
+class _FakeRemoteReference(object):
```

Comment:
I'm sure there must already be at least one implementation of this.... ah, of the five or so in Tahoe, they're all mixed in with other application-specific logic and the one in Foolscap itself is tied to the reactor via `eventually`... so none actually usable.  Swell.

How about doing this with composition instead of inheritance, though?